components/lcd/esp_lcd_gc9503/include/esp_lcd_gc9503.h

```diff
@@ -10,26 +10,63 @@
 
 #pragma once
 
-#include "esp_lcd_types.h"
+#include <stdint.h>
+
+#include "hal/lcd_types.h"
+#include "esp_lcd_panel_vendor.h"
 #include "esp_lcd_panel_rgb.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief Create LCD panel for GC9503.
+ * @brief LCD panel initialization commands.
+ *
+ */
+typedef struct {
+    int cmd;                /*<! The specific LCD command */
+    const void *data;       /*<! Buffer that holds the command specific data */
+    size_t data_bytes;      /*<! Size of `data` in memory, in bytes */
+    unsigned int delay_ms;  /*<! Delay in milliseconds after this command */
+} gc9503_lcd_init_cmd_t;
+
+/**
+ * @brief LCD panel vendor configuration.
+ *
+ * @note  This structure needs to be passed to the `vendor_config` field in `esp_lcd_panel_dev_config_t`.
+ *
+ */
+typedef struct {
+    const esp_lcd_rgb_panel_config_t *rgb_config;   /*!< RGB panel configuration */
+    const gc9503_lcd_init_cmd_t *init_cmds;         /*!< Pointer to initialization commands array.
+                                                     *   The array should be declared as `static const` and positioned outside the function.
+                                                     *   Please refer to `vendor_specific_init_default` in source file
+                                                     */
+    uint16_t init_cmds_size;    /*<! Number of commands in above array */
+    struct {
+        unsigned int use_external_init_cmds: 1; /*<! Set to 1 if use external initialization commands instead of internal default commands */
```

Comment:
Do we need it? I think, we can only check `if(xxx->init_cmds != NULL)`

Reply:
This flag is indeed not that necessary, but it's worth considering a scenario where it might be needed: if a user doesn't want to initialize the screen, similar to the ST7789 driver in IDF.

Reply:
If a user doesn't want to use external cmds, then put only NULL to `init_cmds` and it takes defaults.

Reply:
What I mean is that users don't want to send any initialization commands, including within the driver itself.

Reply:
This flag has been removed.